### ISSUE TITLE
chart 2.5.1:  Has no PodSecurityPolicy , causing failed pods in new namespaces

### ISSUE THREAD

**Issue by atamgp** [rancher/rancher#29696]
**What kind of request is this (question/bug/enhancement/feature request):**
bug/question

**Steps to reproduce (least amount of steps as possible):**
Upgrade from 2.4.x to 2.5.1 using helm chart

**Result:**
new namespaces are created:
- fleet-system
- rancher-operator-system

The pods here cannot start:
```
Error: container has runAsNonRoot and image will run as root
```

**Other details that may be helpful:**
The chart should have a PodSecurityPolicy .
Also it is not possible to pass a securityContext as value to the chart:
https://github.com/rancher/rancher/blob/2.5-head/chart/values.yaml

We fixed this by using a **privileged** ClusterRoleBinding to the whole namespaces mentioned...

gz#13500

gz#13422

**Comment by jiaqiluo** [comment]
This bug is reproduced on rancher:v2.5.2

Steps:
- follow the documentation to prepare a hardened cluster by using RKE CLI https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.4/hardening-2.4/
- install Rancher v2.5.2 into the cluster 

Results:


```
> k describe pod  -n rancher-operator-system rancher-operator-6659bbb889-55z6l | grep -i error
      Reason:       CreateContainerConfigError
  Warning  Failed     32m (x12 over 34m)     kubelet            Error: container has runAsNonRoot and image will run as root
```

```
>  k describe pod -n fleet-system fleet-controller-5db6bcbb9-lj529 | grep -i error
      Reason:       CreateContainerConfigError
  Warning  Failed     34m (x12 over 36m)   kubelet            Error: container has runAsNonRoot and image will run as root
```

<img width="1428" alt="Screen Shot 2020-11-11 at 4 50 32 PM" src="https://user-images.githubusercontent.com/6218999/98877606-17c08f80-243e-11eb-875f-748f1ab97ef7.png">

**Comment by caiconkhicon** [comment]
I have the same issue when updating from 2.4.5 to 2.5.2

**Comment by jiaqiluo** [comment]
Wait for an RC to validate the changes in the helm chart.

Also, after validating the fix in the dev-v2.5 branch, we should forward port the fix to the master branch.

**Comment by jiaqiluo** [comment]
Failed to validate the fix on rancher:`v2.5.4-rc1` 

It turns out that Rancher `v2.5.4-rc1` helm chart does NOT use the new version of `gitjob` and `fleet`
<img width="1408" alt="Screen Shot 2020-12-04 at 11 54 59 AM" src="https://user-images.githubusercontent.com/6218999/101203522-07a86400-3628-11eb-82d5-0d984872109a.png">

**Comment by StrongMonkey** [comment]
Available to test on v2.5-head

**Comment by atamgp** [comment]
Can someone please test with v2.5.4-rc6?

**Comment by jiaqiluo** [comment]
The bug fix is validated on `v2.5.4-rc6`

Steps:

- follow the [documentation](https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.4/hardening-2.4/) to prepare a hardened cluster by using RKE CLI 
- install Rancher `v2.5.4-rc6` into the cluster

Results:
- rancher is installed successfully 
- fleet is installed successfully,  fleet `0.3.2-rc5` is used 
- 
![image](https://user-images.githubusercontent.com/6218999/102546711-48b36600-4075-11eb-9617-acafa53d15b6.png)
 
<img width="1590" alt="Screen Shot 2020-12-17 at 2 36 52 PM" src="https://user-images.githubusercontent.com/6218999/102546777-67196180-4075-11eb-8da2-257ca796dc41.png">

**Comment by jiaqiluo** [comment]
@caiconkhicon  @atamgp  The fix will be available in the coming Rancher v2.5.4 release.
